### ISSUE TITLE
question about P_charging sum in utils/Simulate_Actions3.py

### ISSUE THREAD

**Issue by lalala215** [georkara/Chargym-Charging-Station#1]
Dear author,

Thanks for your work. This repo is beneficial to my research. But I have a question about this code line.
https://github.com/georkara/Chargym-Charging-Station/blob/78c0ed2271e342d50dcd9930e41d0ab6b2c2a50d/Chargym_Charging_Station/utils/Simulate_Actions3.py#L41

From my point of view, P_charging includes some EV charging actions even if there is no EV at the charging points. For example, if hour=0, there is no car in the charging points(present_cars=0), BOC is also 0. And If the action in this charging point is larger than 0, then the P_charging also shows a positive value in this charging point. And it will be summed as a total charging power, which is inconsistent with reality. Can you explain this?

I am hoping for your reply.
Best regards.

**Comment by georkara** [comment]
> Dear author,
> 
> Thanks for your work. This repo is beneficial to my research. But I have a question about this code line.
> 
> https://github.com/georkara/Chargym-Charging-Station/blob/78c0ed2271e342d50dcd9930e41d0ab6b2c2a50d/Chargym_Charging_Station/utils/Simulate_Actions3.py#L41
> 
> From my point of view, P_charging includes some EV charging actions even if there is no EV at the charging points. For example, if hour=0, there is no car in the charging points(present_cars=0), BOC is also 0. And If the action in this charging point is larger than 0, then the P_charging also shows a positive value in this charging point. And it will be summed as a total charging power, which is inconsistent with reality. Can you explain this?
> 
> I am hoping for your reply. Best regards.

Thank you for finding this useful. If i am getting right, you mean that an if statement should be included to exclude such a case right? If so, i think that you are right. I think that the provided scenario of arrivals already includes at least one arrival at the first time step (hour=0), so i didnt notice this. Correct me if i misunderstood your question. Thank you very much.

**Comment by lalala215** [comment]
Yes! adding an if statement would be useful in that case to exclude the unwanted charging actions.
Also maybe change this line 41 as
` Total_charging = sum(P_charging[np.where(present_cars[:,hour]>0)])`
would be useful.
This means we only add the present car action into the total charging.
